src/yb/util/atomic.h

```diff
@@ -444,6 +445,43 @@ class AtomicTryMutex {
   std::atomic<bool> locked_{false};
 };
 
+class AtomicTryMutexWithTimestamp {
```

Comment:
I was thinking something like

template <class LockType>
class LockWithTimestamp {
  LockType lock_;
  ...
}

When using it, LockWithTimestamp<simple_spinlock> or LockWithTimestamp<AtomicTryMutex> could be defined.

Reply:
I will recommend declare and define this class under src/yb/util/locks.h and src/yb/util/locks.cc